prog/postgres/postgres_resource_nexus.rb

```diff
@@ -8,18 +8,19 @@ class Prog::Postgres::PostgresResourceNexus < Prog::Base
   subject_is :postgres_resource
 
   extend Forwardable
-  def_delegators :postgres_resource, :server
+  def_delegators :postgres_resource, :servers, :representative_server
 
-  semaphore :destroy, :update_firewall_rules
+  semaphore :update_firewall_rules, :refresh_dns_record, :destroy
 
-  def self.assemble(project_id:, location:, name:, target_vm_size:, target_storage_size_gib:, parent_id: nil, restore_target: nil)
+  def self.assemble(project_id:, location:, name:, target_vm_size:, target_storage_size_gib:, ha_type: PostgresResource::HaType::NONE, parent_id: nil, restore_target: nil)
     unless (project = Project[project_id])
       fail "No existing project"
     end
 
-    Validation.validate_vm_size(target_vm_size)
-    Validation.validate_name(name)
     Validation.validate_location(location, project.provider)
+    Validation.validate_name(name)
+    Validation.validate_vm_size(target_vm_size)
+    Validation.validate_postgres_ha_type(ha_type)
```

Comment:
Should we really validate this here? The database will already reject this and the request will fail with a db constraint error. That way, we'll have a single source of truth, DB.

Reply:
Generally `assemble` is called synchronously, though. Is that true in this case, and are you accounting for that in your determination? At first glance I thought this was in a label in which case it'd be rather odd to validate.

 Perhaps a counter-indication of my rather thoughtless authorship of the `self.assemble` pattern.

Reply:
My primary motivation for the check was returning proper error message as well.